src/server/pps/server/api_server.go

```diff
@@ -1822,7 +1814,7 @@ func (a *apiServer) validatePipelineRequest(request *pps.CreatePipelineRequest)
 	return nil
 }
 
-func (a *apiServer) validatePipeline(pachClient *client.APIClient, pipelineInfo *pps.PipelineInfo) error {
+func (a *apiServer) validatePipelineInTransaction(txnCtx *txnenv.TransactionContext, pipelineInfo *pps.PipelineInfo) error {
```

Comment:
a double pointer always just looks wrong to me, but it seemed a lot more natural than making sure an extra return value is included on all the exit paths, even with errors

Reply:
You could have it always pass in a commit, but a zero-value commit is treated the same as a `nil` is now.